src/app/pages/splash/services/splash.service.ts

```diff
@@ -18,7 +19,14 @@ export class SplashService {
   ) {}
 
   evalEnrolment() {
-    return this.token.isValid().catch(() => false)
+    return this.token
+      .refresh()
+      .catch(e => {
+        if (e.error.error_description.includes(this.INVALID_USER_ERROR))
```

Comment:
what is the HTTP code returned? Can we check that first and then check for this string